bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -3602,6 +3627,18 @@ void BCStateTran::reportLastAgreedPrunableBlockId(uint64_t lastAgreedPrunableBlo
   rvbm_->reportLastAgreedPrunableBlockId(lastAgreedPrunableBlockId);
 }
 
+void BCStateTran::triggerPostProcessing() {
+  if (maxPostprocessedBlockId_ != 0) {
+    return;
+  }
+  uint64_t lastReachableBlockId = as_->getLastReachableBlockNum();
+  uint64_t firstRequiredBlockId = psd_->getFirstRequiredBlock();
+  LOG_INFO(logger_, KVLOG(firstRequiredBlockId, lastReachableBlockId));
+  if ((firstRequiredBlockId > 1) && ((firstRequiredBlockId - 1) > (lastReachableBlockId + 1))) {
```

Comment:
I didn't understand condition firstRequiredBlockId - 1.

Reply:
firstRequiredBlockId is a networking non RO-replica persitency parameter, It is the next parameter we will fetch from source. If you look at where it is set, if we set it , it means that it all blocks up to this point (excluding) were fetched + committed already, but not linked. If we crashed, this is the absolute minimal upper bounry where post processing can try to link.
In practice, the linkage will be done in kv blockchain level on ctor. here we just trigger the processor thread.
It should link from last reachble block + 1 until firstRequiredBlockId-1 at minimum. It might try to link more.

Reply:
Thanks Gil. 
I think I should have written both comments in specific words. 
It unnecessarily made you write so many things.